analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseReturnStatement.vb

```diff
@@ -0,0 +1,113 @@
+
+    Public Shared Function SharedFunction() As Integer
+        SharedFunction = 42  ' Noncompliant
+    End Function
+    
+    Public Function ExplictlyReturnDefaultReturnValue() As Integer
+        Return ExplictlyReturnDefaultReturnValue ' Noncompliant {{Do not make use of the implicit return value.}}
+    End Function
+        
+    Public Function ReadAssignementStatements() As Integer
+        dim value as Integer = ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        value += ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        value -= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        value *= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        value /= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        value \= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        value &= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        value ^= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        value <<= ReadAssignementStatements ' Noncompliant {{Do not make use of the implicit return value.}}
+        value >>= ReadAssignementStatements ' Noncompliant {{Do not make use of the implicit return value.}}
+        Return value
+    End Function
+
+End Class
+
+Public Class DoesNotApplyOn
+    Implements IInterface
+
+    Public Function FunctionWithExplictReturnOnly()
+        Return 42 'Compliant
+    End Function
+
+    Public Sub SubMethods(number As Integer)
+        Dim SubMethods = number ' Compliant
+    End Sub
+
+    Public Function RecursiveFunctionCalls(number As Integer)
+        If number = 42 Then
+            Return 42
+        Else
+            Return RecursiveFunctionCalls(42) ' Compliant, method call
+        End If
+    End Function
+    
+    Public Function CallToOtherProperty(other As OtherType) As Integer
+        With other
+            Return .CallToOtherProperty ' Compliant
+        End With
+    End Function
+
+    Public Function CallToOtherFunction(other As OtherType) As Integer
+        With other
+            Return .CallToOtherFunction() ' Compliant
+        End With
+    End Function
+
+    Public Function WriteToOtherProperty() As OtherType
+        Return New OtherType With
+        {
+            .WriteToOtherProperty = 69 ' Compliant
+        }
+    End Function
+    
+    Public Function WriteToOtherFunction(other As OtherType) As Integer
+        With other
+            Return .WriteToOtherFunction(42) ' Compliant
+        End With
+    End Function
+    
+    Public Function ImplementedInterfaceMethod() As Integer Implements IInterface.ImplementedInterfaceMethod ' Compliant
+        Return 42
+    End Function
```

Comment:
Let's add 2nd method with noncompliant to be sure our stop condition doesn't kill that

Reply:
Done.